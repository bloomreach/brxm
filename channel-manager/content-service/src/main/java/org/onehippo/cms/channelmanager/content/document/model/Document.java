/*
 * Copyright 2016-2019 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.onehippo.cms.channelmanager.content.document.model;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * This bean represents a document, stored in the CMS. It can be serialized into JSON to expose it through a REST API.
 */
@JsonInclude(Include.NON_NULL)
public class Document {
    private String id;                // UUID
    private String variantId;
    private String branchId;
    private String displayName;
    private String urlName;
    private String repositoryPath;
    private DocumentInfo info;        // read-only information about (the current state of) the document
    private Map<String, List<FieldValue>> fields;
    private String state;

    public Document() {
        setInfo(new DocumentInfo());
        setFields(new HashMap<>());
    }

    public String getId() {
        return id;
    }

    public void setId(final String id) {
        this.id = id;
    }

    public String getVariantId() {
        return variantId;
    }

    public void setVariantId(final String variantId) {
        this.variantId = variantId;
    }

    public String getBranchId() {
        return branchId;
    }

    public void setBranchId(final String branchId) {
        this.branchId = branchId;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(final String displayName) {
        this.displayName = displayName;
    }

    public String getUrlName() {
        return urlName;
    }

    public void setUrlName(final String urlName) {
        this.urlName = urlName;
    }

    public String getRepositoryPath() {
        return repositoryPath;
    }

    public void setRepositoryPath(final String repositoryPath) {
        this.repositoryPath = repositoryPath;
    }

    public DocumentInfo getInfo() {
        return info;
    }

    public void setInfo(final DocumentInfo info) {
        this.info = info;
    }

    public Map<String, List<FieldValue>> getFields() {
        return fields;
    }

    public void setFields(final Map<String, List<FieldValue>> fields) {
        this.fields = fields;
    }

    public String getState() {
        return state;
    }

    public void setState(final String state) {
        this.state = state;
    }
}
