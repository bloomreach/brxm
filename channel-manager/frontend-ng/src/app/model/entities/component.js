/*
 * Copyright 2020 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as HstConstants from '../constants';
import { ComponentEntity } from './component-entity';

export class Component extends ComponentEntity {
  hasExperiment() {
    return !!this.getExperimentId();
  }

  getExperimentId() {
    return this._meta[HstConstants.EXPERIMENT_ID];
  }

  getExperimentStateLabel() {
    if (!this.hasExperiment()) {
      return null;
    }

    return `EXPERIMENT_LABEL_${this._meta[HstConstants.EXPERIMENT_STATE]}`;
  }

  getContainer() {
    return this.container;
  }

  setContainer(container) {
    this.container = container;
  }

  getType() {
    return 'component';
  }

  getRenderVariant() {
    return this._meta[HstConstants.RENDER_VARIANT] || HstConstants.DEFAULT_RENDER_VARIANT;
  }

  getReferenceNamespace() {
    return this._meta[HstConstants.REFERENCE_NAMESPACE];
  }
}
