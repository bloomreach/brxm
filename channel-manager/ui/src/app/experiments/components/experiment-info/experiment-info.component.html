<!--
  Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
  -->

<em-experiment-info-bar *ngIf="isExperimentCreated">
  {{ 'EXPERIMENT_INFO_PUBLISH_TO_START' | translate }}
</em-experiment-info-bar>
<em-experiment-info-bar *ngIf="isExperimentCompleted">
  {{ 'EXPERIMENT_LABEL_COMPLETED' | translate }}.
  {{ 'EXPERIMENT_PUBLISH_TO_FINISH' | translate }}
</em-experiment-info-bar>

<mat-list role="list" class="small-list">
  <mat-list-item role="listitem">
    <span class="experiment-goal-header">{{ experiment | experimentName }}</span>
    <mat-divider></mat-divider>
  </mat-list-item>
  <mat-list-item role="listitem">
    <span class="line-header">{{ 'EXPERIMENT_GOAL' | translate }}</span>
    <span>{{ experiment.goal.name }}</span>
    <mat-divider></mat-divider>
  </mat-list-item>
  <mat-list-item role="listitem">
    <span class="line-header">{{ 'EXPERIMENT_PAGE' | translate }}</span>
    <span>{{ experiment.goal.targetPage }}</span>
    <mat-divider></mat-divider>
  </mat-list-item>
  <mat-list-item role="listitem">
    <span class="line-header">{{ 'EXPERIMENT_STATE' | translate }}</span>
    <span class="experiment-state">{{ experiment.state | lowercase }}</span>
    <mat-divider></mat-divider>
  </mat-list-item>
  <mat-list-item *ngIf="experiment.startTime > 0" role="listitem">
    <span class="line-header">{{ 'EXPERIMENT_START_DATE' | translate }}</span>
    <span>{{ experiment.startTime | moment:'LL' }}</span>
    <mat-divider></mat-divider>
  </mat-list-item>
</mat-list>
