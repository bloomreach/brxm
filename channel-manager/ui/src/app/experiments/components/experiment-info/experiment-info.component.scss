/*
 * Copyright 2020-2023 Bloomreach
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '~@bloomreach/frontend-theme/sizes';
@import 'variables';

$mat-list-line-header-size-with-padding: 110px;

:host {
  display: block;
}

.mat-list-item {
  font-size: $font-normal;

  .experiment-goal-header {
    font-weight: bold;
  }

  .line-header {
    flex-basis: $mat-list-line-header-size-with-padding;
  }

  .experiment-state {
    text-transform: capitalize;
  }

  .mat-divider {
    left: 16px;
    right: 16px;
    width: auto;
  }
}

.small-list .mat-list-item {
  height: $br-size-xl;
}
