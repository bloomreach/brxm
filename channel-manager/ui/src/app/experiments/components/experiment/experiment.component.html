<!--
  Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
  -->

<ng-container *ngIf="experiment$ | async as experiment; else experimentForm">
  <em-experiment-info [experiment]="experiment"></em-experiment-info>

  <div class="experiment-status-wrapper">
    <em-experiment-status-message [winnerVariant]="experiment.winnerVariant"
                                  [visits]="experiment.totalVisits"
                                  class="qa-experiment-status-message">
    </em-experiment-status-message>
    <em-experiment-status-chart [experiment]="experiment"></em-experiment-status-chart>
  </div>
</ng-container>

<ng-template #experimentForm>
  <em-experiment-start-form [variants]="availableVariants$ | async"
                            [goals]="availableGoals$ | async">
  </em-experiment-start-form>
</ng-template>
