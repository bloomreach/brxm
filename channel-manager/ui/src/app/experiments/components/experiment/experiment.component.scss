/*
 * Copyright 2020-2023 Bloomreach
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '~@bloomreach/frontend-theme/sizes';

.experiment {
  height: 100%;
  display: grid;
  grid-template-rows: auto 1fr auto;
  grid-template-areas:
    "info"
    "status"
    "actions";
}

em-experiment-info {
  grid-area: info;
}

.experiment-status-wrapper {
  grid-area: status;
  padding: $br-size;
}

.actions-wrapper {
  grid-area: actions;
}

.actions {
  padding: $br-size-m $br-size;
  text-align: right;
}

.variants-info {
  list-style-type: '- ';
  padding: 0;

  > li {
    padding-top: $br-size-s;
  }
}
