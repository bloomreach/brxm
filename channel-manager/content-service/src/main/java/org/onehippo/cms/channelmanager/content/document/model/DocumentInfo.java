/*
 * Copyright 2016-2020 Hippo B.V. (http://www.onehippo.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.onehippo.cms.channelmanager.content.document.model;

import java.util.List;
import java.util.Objects;

import org.onehippo.cms.channelmanager.content.documenttype.model.DocumentType;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * This bean carries information of a document, stored in the CMS.
 * It is part of a document and can be serialized into JSON to expose it through a REST API.
 * Type {@code type} attribute refers to the document's {@link DocumentType} by id.
 */
@JsonInclude(JsonInclude.Include.NON_DEFAULT)
// only return the error count and error messages, never let clients set it
@JsonIgnoreProperties(value = {"errorCount", "errorMessages"}, allowGetters = true)
public class DocumentInfo {

    /**
     * enveloped reference to document type: { id: "namespace:typename" }
     */
    private Type type;

    // whether this document has auto-drafted changes that have not been saved to the preview variant yet
    private boolean dirty;

    // the number of validation errors
    private int errorCount;

    // human-readable messages of document-level validation errors
    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    private List<String> errorMessages;

    // maps to hippostd:publishableSummary (new, live, changed, or unknown)
    private PublicationState publicationState;

    private boolean canPublish;
    private boolean canRequestPublication;
    private String locale;
    private boolean canKeepDraft;
    private boolean retainable;

    public boolean isRetainable() {
        return retainable;
    }

    public Type getType() {
        return type;
    }

    public void setTypeId(final String id) {
        type = new Type(id);
    }

    public void setCanKeepDraft(final boolean saveDraft) {
        this.canKeepDraft = saveDraft;
    }

    public void setRetainable(final boolean retainable) {
        this.retainable = retainable;
    }

    public static class Type {

        private final String id;

        @JsonCreator
        public Type(@JsonProperty("id") String id) {
            this.id = id;
        }

        public String getId() {
            return id;
        }
    }

    public boolean isDirty() {
        return dirty;
    }

    public void setDirty(final boolean dirty) {
        this.dirty = dirty;
    }

    public int getErrorCount() {
        return errorCount;
    }

    public void setErrorCount(final int errorCount) {
        this.errorCount = errorCount;
    }

    public List<String> getErrorMessages() {
        return errorMessages;
    }

    public void setErrorMessages(final List<String> errorMessages) {
        this.errorMessages = errorMessages;
    }

    public PublicationState getPublicationState() {
        return publicationState;
    }

    public void setPublicationState(final PublicationState publicationState) {
        this.publicationState = publicationState;
    }

    public boolean isCanPublish() {
        return canPublish;
    }

    public boolean isCanKeepDraft(){
        return canKeepDraft;
    }

    public boolean isCanRequestPublication() {
        return canRequestPublication;
    }

    public void setCanPublish(final boolean canPublish) {
        this.canPublish = canPublish;
    }

    public void setCanRequestPublication(final boolean canRequestPublication) {
        this.canRequestPublication = canRequestPublication;
    }

    public String getLocale() {
        return locale;
    }

    public void setLocale(final String locale) {
        this.locale = locale;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof DocumentInfo)) {
            return false;
        }
        final DocumentInfo that = (DocumentInfo) o;
        return isDirty() == that.isDirty() &&
                getErrorCount() == that.getErrorCount() &&
                isCanPublish() == that.isCanPublish() &&
                isCanRequestPublication() == that.isCanRequestPublication() &&
                isCanKeepDraft() == that.isCanKeepDraft() &&
                isRetainable() == that.isRetainable() &&
                Objects.equals(getType(), that.getType()) &&
                Objects.equals(getErrorMessages(), that.getErrorMessages()) &&
                getPublicationState() == that.getPublicationState() &&
                Objects.equals(getLocale(), that.getLocale());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getType(), isDirty(), getErrorCount(), getErrorMessages(), getPublicationState(), isCanPublish(), isCanRequestPublication(), getLocale(), isCanKeepDraft(), isRetainable());
    }


    @Override
    public String toString() {
        return "DocumentInfo{" +
                "type=" + type +
                ", dirty=" + dirty +
                ", errorCount=" + errorCount +
                ", errorMessages=" + errorMessages +
                ", publicationState=" + publicationState +
                ", canPublish=" + canPublish +
                ", canRequestPublication=" + canRequestPublication +
                ", locale='" + locale + '\'' +
                ", canKeepDraft=" + canKeepDraft +
                ", retainable=" + retainable +
                '}';
    }
}
