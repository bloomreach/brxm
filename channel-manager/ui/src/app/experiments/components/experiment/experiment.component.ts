/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Component, Inject } from '@angular/core';
import { TranslateService } from '@ngx-translate/core';

import { Ng1ComponentEditorService, NG1_COMPONENT_EDITOR_SERVICE } from '../../../services/ng1/component-editor.ng1.service';
import { NotificationService } from '../../../services/notification.service';
import { VariantsService } from '../../../variants/services/variants.service';
import { ExperimentState } from '../../models/experiment-state.enum';
import { Experiment } from '../../models/experiment.model';
import { ExperimentsService } from '../../services/experiments.service';
import { SelectedVariantIdAndGoalId } from '../experiment-start-form/experiment-start-form.component';

@Component({
  templateUrl: 'experiment.component.html',
  styleUrls: ['experiment.component.scss'],
})
export class ExperimentComponent {
  private readonly component = this.componentEditorService.getComponent();
  private readonly componentId = this.component.getId();
  readonly availableVariants$ = this.variantsService.getVariants(this.componentId);
  readonly availableGoals$ = this.experimentsService.getGoals();

  experiment$ = this.experimentsService.getExperiment(this.componentId);

  requestInProgress = false;

  constructor(
    @Inject(NG1_COMPONENT_EDITOR_SERVICE) private readonly componentEditorService: Ng1ComponentEditorService,
    private readonly experimentsService: ExperimentsService,
    private readonly variantsService: VariantsService,
    private readonly notificationService: NotificationService,
    private readonly translateService: TranslateService,
  ) {}

  isExperimentRunning(experiment: Experiment): boolean {
    return experiment.state === ExperimentState.Running;
  }

  isExperimentRunningOrCompleted(experiment: Experiment): boolean {
    return this.isExperimentRunning(experiment) || experiment.state === ExperimentState.Completed;
  }

  async onVariantAndGoalSelected(value: SelectedVariantIdAndGoalId): Promise<void> {
    const { variantId, goalId } = value;

    try {
      this.requestInProgress = true;

      await this.experimentsService.saveExperiment(this.componentId, variantId, goalId);

      this.experiment$ = this.experimentsService.getExperiment(this.componentId);

      this.notificationService.showNotification(this.translateService.instant('EXPERIMENT_SAVED'));
    } catch (e) {
      this.notificationService.showErrorNotification(this.translateService.instant('EXPERIMENT_SAVE_ERROR'));
    } finally {
      this.requestInProgress = false;
    }
  }

  async onCompleteExperiment(experiment: Experiment): Promise<void> {
    const firstNonDefaultVariant = experiment.variants.find(v => v.variantId !== this.variantsService.defaultVariantId);

    if (!firstNonDefaultVariant) {
      throw new Error('Unable to find a non default variant');
    }

    try {
      this.requestInProgress = true;

      await this.experimentsService.completeExperiment(this.componentId, firstNonDefaultVariant.variantId);

      this.experiment$ = this.experimentsService.getExperiment(this.componentId);

      this.notificationService.showNotification(this.translateService.instant('EXPERIMENT_COMPLETED'));
    } catch {
      this.notificationService.showErrorNotification(this.translateService.instant('EXPERIMENT_COMPLETION_ERROR'));
    } finally {
      this.requestInProgress = false;
    }
  }
}
