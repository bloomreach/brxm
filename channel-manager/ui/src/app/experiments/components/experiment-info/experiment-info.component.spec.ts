/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { NO_ERRORS_SCHEMA, Pipe, PipeTransform } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ExperimentState } from '../../models/experiment-state.enum';
import { Experiment } from '../../models/experiment.model';

import { ExperimentInfoComponent } from './experiment-info.component';

@Pipe({name: 'experimentName'})
export class ExperimentNameMockPipe implements PipeTransform {
  transform(value: Experiment): string {
    return value.id.toString();
  }
}

@Pipe({name: 'translate'})
export class TranslateMockPipe implements PipeTransform {
  transform(value: string): string {
    return value;
  }
}

@Pipe({name: 'moment'})
export class MomentMockPipe implements PipeTransform {
  transform(value: number | string | Date, format?: string): string {
    return `${value}`;
  }
}

describe('ExperimentInfoComponent', () => {
  let fixture: ComponentFixture<ExperimentInfoComponent>;
  let component: ExperimentInfoComponent;

  const mockExperiment: Experiment = {
    id: 'experiment-1',
    state: ExperimentState.Unknown,
    type: 'PAGE',
    startTime: 1600868196063,
    winnerVariant: null,
    goal: {
      id: 'goal-1',
      name: 'Goal 1',
      type: 'PAGE',
      readOnly: false,
      targetPage: '/target-page',
      mountId: 'some-mount-id',
    },
    variants: [
      {
        variantId: 'default',
        variantName: 'Default',
        confidence: 0.3,
        mean: 0.2,
        variance: 0.4,
        visitorSegment: 'segment-1',
      },
      {
        variantId: 'some-variant',
        variantName: 'Some variant',
        confidence: 0.3,
        mean: 0.2,
        variance: 0.4,
        visitorSegment: 'segment-1',
      },
    ],
  };

  beforeEach(() => {
    fixture = TestBed.configureTestingModule({
      declarations: [
        ExperimentInfoComponent,
        ExperimentNameMockPipe,
        TranslateMockPipe,
        MomentMockPipe,
      ],
      schemas: [
        NO_ERRORS_SCHEMA,
      ],
    }).createComponent(ExperimentInfoComponent);

    component = fixture.componentInstance;

    component.experiment = mockExperiment;
  });

  describe.each([
    ExperimentState.Created,
    ExperimentState.Running,
    ExperimentState.Completed,
    ExperimentState.Unknown,
  ])('%s experiment', state => {
    beforeEach(() => {
      component.experiment = {
        ...mockExperiment,
        state,
      };

      if (state === ExperimentState.Created) {
        component.experiment.startTime = -1;
      }

      fixture.detectChanges();
    });

    test('should be shown', () => {
      expect(fixture.nativeElement).toMatchSnapshot();
    });
  });
});
