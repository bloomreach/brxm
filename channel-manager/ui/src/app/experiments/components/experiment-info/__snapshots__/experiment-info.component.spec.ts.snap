// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperimentInfoComponent COMPLETED experiment should be shown 1`] = `
<div
  id="root2"
>
  
  
  <em-experiment-info-bar>
     EXPERIMENT_LABEL_COMPLETED. EXPERIMENT_PUBLISH_TO_FINISH

  </em-experiment-info-bar>
  <mat-list
    class="small-list"
    role="list"
  >
    <mat-list-item
      role="listitem"
    >
      <span
        class="experiment-goal-header"
      >
        experiment-1
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_GOAL
      </span>
      <span>
        Goal 1
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_PAGE
      </span>
      <span>
        /target-page
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_STATE
      </span>
      <span
        class="experiment-state"
      >
        completed
      </span>
      <mat-divider />
    </mat-list-item>
    
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_START_DATE
      </span>
      <span>
        1600868196063
      </span>
      <mat-divider />
    </mat-list-item>
  </mat-list>
</div>
`;

exports[`ExperimentInfoComponent CREATED experiment should be shown 1`] = `
<div
  id="root0"
>
  
  <em-experiment-info-bar>
     EXPERIMENT_INFO_PUBLISH_TO_START

  </em-experiment-info-bar>
  
  <mat-list
    class="small-list"
    role="list"
  >
    <mat-list-item
      role="listitem"
    >
      <span
        class="experiment-goal-header"
      >
        experiment-1
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_GOAL
      </span>
      <span>
        Goal 1
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_PAGE
      </span>
      <span>
        /target-page
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_STATE
      </span>
      <span
        class="experiment-state"
      >
        created
      </span>
      <mat-divider />
    </mat-list-item>
    
  </mat-list>
</div>
`;

exports[`ExperimentInfoComponent RUNNING experiment should be shown 1`] = `
<div
  id="root1"
>
  
  
  <mat-list
    class="small-list"
    role="list"
  >
    <mat-list-item
      role="listitem"
    >
      <span
        class="experiment-goal-header"
      >
        experiment-1
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_GOAL
      </span>
      <span>
        Goal 1
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_PAGE
      </span>
      <span>
        /target-page
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_STATE
      </span>
      <span
        class="experiment-state"
      >
        running
      </span>
      <mat-divider />
    </mat-list-item>
    
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_START_DATE
      </span>
      <span>
        1600868196063
      </span>
      <mat-divider />
    </mat-list-item>
  </mat-list>
</div>
`;

exports[`ExperimentInfoComponent UNKNOWN experiment should be shown 1`] = `
<div
  id="root3"
>
  
  
  <mat-list
    class="small-list"
    role="list"
  >
    <mat-list-item
      role="listitem"
    >
      <span
        class="experiment-goal-header"
      >
        experiment-1
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_GOAL
      </span>
      <span>
        Goal 1
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_PAGE
      </span>
      <span>
        /target-page
      </span>
      <mat-divider />
    </mat-list-item>
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_STATE
      </span>
      <span
        class="experiment-state"
      >
        unknown
      </span>
      <mat-divider />
    </mat-list-item>
    
    <mat-list-item
      role="listitem"
    >
      <span
        class="line-header"
      >
        EXPERIMENT_START_DATE
      </span>
      <span>
        1600868196063
      </span>
      <mat-divider />
    </mat-list-item>
  </mat-list>
</div>
`;
